hooks/useApproval.ts

```diff
@@ -11,55 +18,66 @@ const APPROVE_CALLBACK_STATUS = {
 const useApprove = (erc20: string, spender: string) => {
   const [status, setStatus] = useState<number>()
   const { data: signer } = useSigner()
-  const contract = useContract({
-    addressOrName: erc20,
-    contractInterface: erc20ABI,
-    signerOrProvider: signer,
-  })
+  const { write: contractApprove } = useContractWrite(
+    {
+      addressOrName: erc20,
+      contractInterface: erc20ABI,
+      signerOrProvider: signer
+    },
+    'approve',
+    {
+      args: [spender, ethers.constants.MaxUint256],
+      onError(error) {
+        console.log(error)
+        setStatus(APPROVE_CALLBACK_STATUS.FAIL)
+      },
+      onSuccess(data) {
+        setStatus(APPROVE_CALLBACK_STATUS.PENDING)
+      }
+    })
 
   return {
     status: status,
-    approve: () => contract.approve(
-      spender,
-      ethers.constants.MaxUint256
-    )
-      .then(() => { setStatus(APPROVE_CALLBACK_STATUS.PENDING) })
-      .catch(() => { setStatus(APPROVE_CALLBACK_STATUS.FAIL) })
+    approve: () => contractApprove()
   }
 }
 
 const useApproval = (erc20: string, spenderAddress: string) => {
   const [approved, setApprove] = useState<boolean>()
```

Comment:
```suggestion
  const [approved, setApprove] = useState<boolean>(false)
```

If unnecessary, it is usually easier to avoid undefined for type definitions.